src/avro/src/schema.rs

```diff
@@ -2067,6 +2068,12 @@ fn field_ordering_position(field: &str) -> Option<usize> {
         "items" => 5,
         "values" => 6,
         "size" => 7,
+        // Supports decimals
```

Comment:
This violates https://avro.apache.org/docs/current/spec.html#Transforming+into+Parsing+Canonical+Form , which specifies that everything except name/type/fields/symbols/items/values/size should be stripped when putting a schema in canonical form.

Perhaps we just shouldn't be canonicalizing before putting things in schema registry?

Reply:
Ah, maybe. It seems like the canonicalization spec is incomplete because that same doc [says](https://avro.apache.org/docs/current/spec.html#Decimal):

> For the purposes of schema resolution, two schemas that are decimal logical types match if their scales and precisions match.

The doc also implies that schema resolution is "valid" [only for canonical schemas](https://avro.apache.org/docs/current/spec.html#Parsing+Canonical+Form+for+Schemas):

> Parsing Canonical Form is a transformation of a writer's schema that let's us define what it means for two schemas to be "the same" for the purpose of reading data written against the schema.

So implicit in this is that scale and precision must be present, but they aren't explicitly part of canonicalization. I defer to your preference in handling this.

Reply:
yep, the doc is confusing. It seems to suggest that _whether to even do_ resolution is determined by whether the canonical forms are equal, but resolution itself requires the non-canonicalized forms! I suspect this is a mistake (or at least that it could be clarified substantially) and I can write to the Avro mailing list about it.

I think the fix for now is not to change how we do canonicalization, but rather to just not canonicalize the schemas before we write them to CSR. I think we had to make the same fix in Materialize a while back to get decimals to work for a customer.

Reply:
I.e., get rid of the `canonical_form` calls in kgen.rs

Reply:
ty for the pointer; removing `canonical_form` calls here fixed the issue.